src/Test/TestAIMuriel.java

```diff
@@ -289,17 +329,52 @@ private boolean testTankWadeThroughInfs()
       {
         act = testCo1.getNextAIAction(testMap);
         if( null != act )
-          testPassed &= validate(performGameAction(act, testGame), "    Muriel generated a bad action!");
+          testPassed &= validate(performGameAction(act, testGame), "    "+ai.getName()+" generated a bad action!");
       } while( null != act && testPassed );
     }
 
-    testPassed &= validate(nmeArty.getHP() <= 0, "    Muriel failed to kill the artillery!");
-    testPassed &= validate(testMap.getResident(9, 0).CO == inf.CO, "    Muriel failed to move infantry onto HQ!");
-    testPassed &= validate(testMap.getLocation(9, 0).getOwner() == inf.CO, "    Muriel failed to capture the HQ!");
+    testPassed &= validate(nmeArty.getHP() <= 0, "    "+ai.getName()+" failed to kill the artillery!");
+    testPassed &= validate(testMap.getResident(9, 0).CO == inf.CO, "    "+ai.getName()+" failed to move infantry onto HQ!");
+    testPassed &= validate(testMap.getLocation(9, 0).getOwner() == inf.CO, "    "+ai.getName()+" failed to capture the HQ!");
 
     // Clean up
     cleanupTest();
     return testPassed;
   }
 
+  /** Confirm that the AI will clear its factory to build a counter, when there is only one possible counter. */
+  private boolean testProductionClearing(AIMaker ai)
+  {
+    setupTest(MapReader.readSingleMap("src/Test/TestProductionClearing.map"), ai);
```

Comment:
This map does not exist. Not sure we want to create a bunch of map files just for simple test cases, but it might be amusing to have them as actual files so people can try them too.

Reply:
...I coulda sworn I added it. Derp.
IMO it's a lot less effort (and a lot clearer) to use our map format than to hand-code the toy maps.
Encodes terrain/ownership/units in a concise and readable format.

Reply:
I guess. Not sure if we really want a bunch of tiny maps hanging around--but then that might make it easier to let a human try out the AI tests. That could be fun, especially once we make some more complex ones.

Also not sure why you created this test case at all since it's basically a slightly more complicated version of the other megatank test, but eh.

Reply:
> I guess. Not sure if we really want a bunch of tiny maps hanging around

The test maps will be in the test directory, not the general maps directory (unless we really want that).
The map *will* be defined somewhere, one way or the other. The question is basically how readable/reusable it is.